### PR TITLE
[MMI] custody page component

### PR DESCRIPTION
**Pull request by zone-live**
## Explanation

This PR adds the MMI `CustodyPage` under the `pages/institutional/` folder and moves the `CustodyAccountList` page under the same folder.
Ref ticket: https://consensyssoftware.atlassian.net/browse/MMI-2650

Screenshot 2023-04-20 at 15 41 36

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue

### CODE REVIEW

File: ui/pages/institutional/custody/custody.js [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+    setSelectedCustodianType('');
+    setSelectedCustodianImage(null);
+    setSelectedCustodianDisplayName('');
+    setApiUrl('');
+    setCurrentJwt('');
+    setConnectError('');
+    setSelectError('');
+  };
+
+  const renderConnectButton = () => {
```
Comment (brad-decker) [inline ui/pages/institutional/custody/custody.js:387]: [REQ] Please remove all these renderMethods. Just render them in the JSX that is returned at the component level. As some advice, if you find yourself reaching for abstracting parts of your tree into separate functions you are likely either over thinking the complexity of the JSX or the component you are making is too large and should be further broken into sub pieces that are their own files and own components.

File: ui/pages/institutional/custody/custody.js [hunk 2]
```diff
@@ -0,0 +1,677 @@
...
+  const [currentJwt, setCurrentJwt] = useState('');
+  const [selectError, setSelectError] = useState('');
+  const [jwtList, setJwtList] = useState([]);
+  const [apiUrl, setApiUrl] = useState('');
+  const [addNewTokenClicked, setAddNewTokenClicked] = useState(false);
+  const [chainId, setChainId] = useState(0);
+  const [connectRequest, setConnectRequest] = useState(undefined);
+  const [accounts, setAccounts] = useState();
+
+  const handleConnectError = useCallback(
```
Comment (brad-decker) [inline ui/pages/institutional/custody/custody.js:79]: I'm not asking for a change here i'm just curious what led to the usage of useCallback and useMemo in this component

File: ui/pages/institutional/custody/index.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+.custody-accounts-empty {
+  min-height: 300px;
+
+  &__title {
+    font-size: 14px;
+  }
+
+  &__subtitle {
+    font-size: 12px;
+  }
```
Comment (brad-decker) [inline ui/pages/institutional/custody/index.scss:17]: Use Text components and text variant sizing not font-size css

File: ui/pages/institutional/custody/index.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
+@import '../../../components/institutional/jwt-dropdown/jwt-dropdown.scss';
+@import '../../../components/institutional/jwt-url-form/jwt-url-form.scss';
+
+.custody-select-all {
+  font-size: 12px;
+}
```
Comment (brad-decker) [inline ui/pages/institutional/custody/index.scss:6]: Use Text Component variant sizing not font size css

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/institutional/custody/custody.js (hunk 1 above)
Comment (zone-live) [reply ui/pages/institutional/custody/custody.js:387]: Will do 👍🏼

File: ui/pages/institutional/custody/custody.js (hunk 2 above)
Comment (zone-live) [reply ui/pages/institutional/custody/custody.js:79]: Lint was giving warnings, telling to add it :)

File: ui/pages/institutional/custody/index.scss (hunk 3 above)
Comment (zone-live) [reply ui/pages/institutional/custody/index.scss:17]: will do 👍🏼

File: ui/pages/institutional/custody/index.scss (hunk 4 above)
Comment (zone-live) [reply ui/pages/institutional/custody/index.scss:6]: will do 👍🏼

File: ui/pages/institutional/custody/custody.js [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+          custodyName: selectedCustodianName,
+        };
+      });
+      setSelectedAccounts(allAccounts);
+    } else {
+      setSelectedAccounts({});
+    }
+  };
+
+  const renderCustodyContent = () => {
```
Comment (brad-decker) [inline ui/pages/institutional/custody/custody.js:305]: Missed removing one of 'renderX' methods.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/institutional/custody/custody.js [hunk 6]
```diff
@@ -0,0 +1,645 @@
...
+            }}
+          >
+            {t('select')}
+          </Button>
+        </Box>,
+      );
+    });
+
+    setCustodianButtons(custodianItems);
+  }, [connectRequest, custodians, dispatch, selectedCustodianName]);
```
Comment (brad-decker) [inline ui/pages/institutional/custody/custody.js:148]: Why is this an effect that sets state instead of useMemo to memorize an array of items?

Comment (brad-decker) [reply ui/pages/institutional/custody/custody.js:148]: In general setting a react tree in state is a bad practice.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/institutional/custody/custody.js [hunk 7]
```diff
@@ -0,0 +1,648 @@
...
+        </Text>
+      )}
+
+      {selectError && (
+        <Text textAlign={TEXT_ALIGN.CENTER} marginTop={3} padding={[2, 7, 5]}>
+          {selectError}
+        </Text>
+      )}
+
+      {renderCustodianButtons()}
```
Comment (brad-decker) [inline ui/pages/institutional/custody/custody.js:427]: This function is called exactly once.... and we do not want these render methods as I mentioned before. Please use useMemo or just run the iterator every time like you are doing now without the render method.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/institutional/custody/custody.js (hunk 6 above)
Comment (brad-decker) [reply ui/pages/institutional/custody/custody.js]: As I have said before, lets not set the precedent of breaking up long functions with render methods we already said we don't want to do that before we made the move to functional components.

File: ui/pages/institutional/custody/custody.js (hunk 7 above)
Comment (zone-live) [reply ui/pages/institutional/custody/custody.js]: Added the useMemo 👍🏼

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/institutional/custody/custody.js [hunk 8]
```diff
@@ -0,0 +1,645 @@
...
+        chainId,
+      );
+    }, 750);
+  }, [
+    getCustodianAccountsByAddress,
+    apiUrl,
+    chainId,
+    currentJwt,
+    selectedCustodianType,
+  ]);
```
[the remark this answers is not in the document]
Comment (zone-live) [reply ui/pages/institutional/custody/custody.js:329]: Not important right now, this is something for us to probably delete even more down the road as we don't have the search even.

File: ui/pages/institutional/custody/custody.js [hunk 9]
```diff
@@ -0,0 +1,645 @@
...
+              });
+            }}
+          >
+            {t('select')}
+          </Button>
+        </Box>,
+      );
+    });
+
+    setCustodianButtons(custodianItems);
```
[the remark this answers is not in the document]
Comment (zone-live) [reply ui/pages/institutional/custody/custody.js]: You're right. Fixed 👍🏼

**Review by brad-decker — DISMISSED** [review]

**Comment by zone-live** [comment]
Hey @brad-decker sorry again, need your approval ✅ once more, had to re-run lint and update the BUTTON_TYPE to BUTTON_VARIANT after updating with develop branch.

File: ui/pages/pages.scss [hunk 10]
```diff
@@ -13,7 +13,11 @@
 @import 'connected-sites/index';
 @import 'create-account/index';
 ///: BEGIN:ONLY_INCLUDE_IN(mmi)
```
Comment (albertolive) [inline ui/pages/pages.scss:15]:
```suggestion
///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
```

**Review by albertolive — DISMISSED** [review]

**Review by albertolive — DISMISSED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Review by albertolive — APPROVED** [review]

**Pull request merged**